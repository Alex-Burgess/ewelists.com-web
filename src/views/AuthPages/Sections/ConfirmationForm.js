import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// libs
import { onAuthError, debugError } from "libs/errorLib";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// core components
import Button from "components/CustomButtons/Button.js";
import CustomInput from "components/CustomInput/CustomInput.js";

import styles from "assets/jss/custom/views/authPageStyle.js";
const useStyles = makeStyles(styles);

export default function ConfirmationForm(props) {
  const classes = useStyles();
  const { email, password } = props;

  const [confirmationCode, setConfirmationCode] = useState('');
  const [confirmationError, setConfirmationError] = useState(false);

  const validateConfirmationForm = () => {
    return confirmationCode.length > 0;
  }

  const handleConfirmationSubmit = async event => {
    event.preventDefault();

    try {
      await Auth.confirmSignUp(email, confirmationCode);
      debugError("User confirmed email.");
      await Auth.signIn(email, password);
    } catch (e) {
      onAuthError(e, email);
      setConfirmationError(e.message);
    }
  }

  return (
    <form className={classes.form} onSubmit={handleConfirmationSubmit}>
      <h3 className={classes.title + " " + classes.textCenter}>Confirmation Code</h3>
      <div id="confirmationMessage">
        <p>
          Please check your email for the code.
        </p>
      </div>
      <CustomInput
        labelText="Confirmation Code"
        id="confirmationCode"
        formControlProps={{
          fullWidth: true,
          value: email,
          onChange: event => setConfirmationCode(event.target.value),
          className: classes.inputProps
        }}
      />
      { confirmationError
        ? <div className={classes.error + " " + classes.textCenter}>
            <p>{confirmationError}</p>
          </div>
        : null
      }
      <div className={classes.textCenter}>
        <Button round color="info" type="submit" disabled={!validateConfirmationForm()}>
          Verify
        </Button>
      </div>
    </form>
  );
}

ConfirmationForm.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired
};
