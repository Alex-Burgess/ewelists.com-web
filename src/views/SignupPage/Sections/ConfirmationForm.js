import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import InputAdornment from "@material-ui/core/InputAdornment";
import Icon from "@material-ui/core/Icon";
// core components
import HeaderTransparent from "custom/Header/HeaderTransparent.js";
import FooterTransparent from "custom/Footer/FooterTransparent.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import Button from "components/CustomButtons/Button.js";
import CustomInput from "components/CustomInput/CustomInput.js";
import {imageSize} from 'custom/Image/Image.js';

import styles from "assets/jss/custom/views/signupPageStyle.js";
const useStyles = makeStyles(styles);

const image = "/images/sheep-with-shoes";

export default function ConfirmationForm(props) {
  const classes = useStyles();
  const { email, password } = props;

  const [confirmationCode, setConfirmationCode] = useState('');
  const [confirmationError, setConfirmationError] = useState(false);

  const validateConfirmationForm = () => {
    return confirmationCode.length > 0;
  }

  const handleConfirmationSubmit = async event => {
    event.preventDefault();

    try {
      await Auth.confirmSignUp(email, confirmationCode);
      console.log("User confirmed email.");
      await Auth.signIn(email, password);
    } catch (e) {
      setConfirmationError(e.message);
    }
  }

  return (
    <div>
      <HeaderTransparent isAuthenticated={false} />
      <div
        className={classes.pageHeader}
        style={{
          backgroundImage: "url(" + imageSize(image) + ")",
          backgroundSize: "cover",
          backgroundPosition: "top center"
        }}
      >
        <div className={classes.container}>
          <GridContainer justify="center">
            <GridItem xs={12} sm={12} md={5}>
              <Card className={classes.cardSignup}>
                <h2 className={classes.cardTitle}>Confirmation Code</h2>
                <CardBody>
                  <form className={classes.form} onSubmit={handleConfirmationSubmit}>
                  <div id="confirmationMessage" className={classes.details}>
                    <p>
                      Please check your email for the code.
                    </p>
                  </div>
                  <CustomInput
                      id="confirmationCode"
                      formControlProps={{
                        fullWidth: true,
                        className: classes.customFormControlClasses,
                        value: confirmationCode,
                        onChange: event => setConfirmationCode(event.target.value)
                      }}
                      inputProps={{
                        startAdornment: (
                          <InputAdornment
                            position="start"
                            className={classes.inputAdornment}
                          >
                            <Icon className={classes.inputAdornmentIcon}>
                              lock_outline
                            </Icon>
                          </InputAdornment>
                        ),
                        autoComplete: "off",
                        placeholder: "Confirmation Code..."
                      }}
                    />
                    { confirmationError
                      ? <div id="confirmationError" className={classes.signUpError}>
                          <p>{confirmationError}</p>
                        </div>
                      : null
                    }
                    <div className={classes.textCenter}>
                      <Button round color="info" type="submit" disabled={!validateConfirmationForm()}>
                        Verify
                      </Button>
                    </div>
                  </form>
                </CardBody>
              </Card>
            </GridItem>
          </GridContainer>
        </div>
        <FooterTransparent />
      </div>
    </div>
  );
}

ConfirmationForm.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired
};
