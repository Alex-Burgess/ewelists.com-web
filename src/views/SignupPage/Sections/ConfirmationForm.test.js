import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { Auth } from "aws-amplify";
import ConfirmationForm from './ConfirmationForm';

describe('Signup Page Snapshots', () => {
  test('Renders empty form of signup page', () => {
    const tree = TestRenderer
      .create(<Router><ConfirmationForm email="mock.user@gmail.com" password="12345678" /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Confirmation Form', () => {
  const wrapper = mount(
    <Router>
      <ConfirmationForm email="mock.user@gmail.com" password="12345678" />
    </Router>
  );

  it('Should have 1 inputs', () => {
    expect(wrapper.find('CustomInput').length).toEqual(1);
  });

  it('Should have confirmationCode input', () => {
    expect(wrapper.find('CustomInput[id="confirmationCode"]').length).toEqual(1);
  });

  it('Should have 1 form submit button', () => {
    expect(wrapper.find('form').find('button').length).toEqual(1);
  });

  it('Form submit button should be disabled', () => {
    const submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);
  });

  it('Submit button should be active when all fields populated', () => {
    wrapper.update();
    let submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="confirmationCode"]').simulate('change', {
      target: {
        value: '123456',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(false);
  });
});

describe('Test Confirmation Form Submit', () => {
  const wrapper = mount(
    <Router>
      <ConfirmationForm email="mock.user@gmail.com" password="12345678" />
    </Router>
  );

  it('should display error if code incorrect', () => {
    wrapper.find('input[id="confirmationCode"]').simulate('change', {
      target: {
        value: '123456',
      },
    });

    wrapper.update();

    Auth.confirmSignUp = jest.fn().mockImplementation(() => {
      throw new Error("Invalid verification code provided, please try again.");
    });

    let form = wrapper.find('form');
    form.simulate('submit');
    expect(wrapper.find('#confirmationError').text()).toEqual('Invalid verification code provided, please try again.');
  });
});
