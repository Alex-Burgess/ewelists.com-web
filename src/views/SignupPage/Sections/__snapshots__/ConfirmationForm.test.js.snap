// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page Snapshots Renders empty form of signup page 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-90 makeStyles-transparent-102 makeStyles-fixed-92 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-container-93 MuiToolbar-gutters"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-94"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                className="makeStyles-logo-89"
                src="logo-white.png"
              />
            </a>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="makeStyles-hidden-106 PrivateHiddenCss-smDown-188"
        >
          <div
            className="makeStyles-collapse-107"
          >
            <ul
              className="MuiList-root makeStyles-list-198 makeStyles-mlAuto-214 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-listItem-199 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-201"
                  href="/list-ideas"
                  onClick={[Function]}
                >
                  Gift List Ideas
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-199 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-201"
                  href="/login"
                  onClick={[Function]}
                >
                  Log In
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-199 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-201"
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="makeStyles-pageHeader-22"
    style={
      Object {
        "backgroundImage": "url(sheep-with-shoes.jpg)",
        "backgroundPosition": "top center",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="makeStyles-container-21"
    >
      <div
        className="MuiGrid-root makeStyles-grid-230  MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-grid-334  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-5"
        >
          <div
            className="makeStyles-card-335 makeStyles-cardSignup-23"
          >
            <h2
              className="makeStyles-cardTitle-24"
            >
              Confirmation Code
            </h2>
            <div
              className="makeStyles-cardBody-350"
            >
              <form
                className="makeStyles-form-55"
                onSubmit={[Function]}
              >
                <div
                  id="confirmationMessage"
                >
                  <p>
                    Please check your email for the code.
                  </p>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-365 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-358 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root makeStyles-inputAdornment-53 MuiInputAdornment-positionStart"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root makeStyles-inputAdornmentIcon-54"
                      >
                        lock_outline
                      </span>
                    </div>
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      className="MuiInputBase-input MuiInput-input makeStyles-input-367 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="confirmationCode"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Confirmation Code..."
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="makeStyles-textCenter-52"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-423 makeStyles-info-427 makeStyles-round-450 makeStyles-disabled-447"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="submit"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Verify
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="makeStyles-footer-467 makeStyles-footer-461"
    >
      <div
        className="makeStyles-container-473"
      >
         
        <div>
          <div
            className="makeStyles-left-458"
          >
            <ul
              className="MuiList-root makeStyles-list-457 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-inlineBlock-456 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-455"
                  href="/"
                >
                  Â© 
                  2020
                   ewelists
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-456 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-455"
                  href="/contact"
                >
                  Contact Us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-456 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-455"
                  href="/#about"
                >
                  About us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-456 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-455"
                  href="/privacy"
                >
                  Privacy
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-456 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-455"
                  href="/terms"
                >
                  Terms
                </a>
              </li>
            </ul>
          </div>
          <div
            className="makeStyles-rightLinks-463"
          >
            <ul>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-423 makeStyles-simple-445 makeStyles-justIcon-453"
                  href="https://www.facebook.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-423 makeStyles-simple-445 makeStyles-justIcon-453"
                  href="https://twitter.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-twitter"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-423 makeStyles-simple-445 makeStyles-justIcon-453"
                  href="https://www.instagram.com/ewelists/"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-instagram"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="makeStyles-clearFix-483"
        />
      </div>
    </footer>
  </div>
</div>
`;
